### ISSUE TITLE
Custom qubes-app-shutdown-idle timeout

### ISSUE THREAD

**Issue by crat0z** [QubesOS/qubes-issues#7458]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)
(I can't come up with a descriptive title, so please feel free to change it)

### The problem you're addressing (if any)
One of the main ideas behind Qubes is to provide security through isolation and compartmentalization. For example, a user might have a VM where they do online banking, another where they use chat programs, another where they hold GPG keys etc. Currently (IMO) it can be a little tedious to use VMs in this manner because the user has to remember to shutdown the VM after using it. When a user forgets to shutdown the VM, they are wasting RAM, which is a very important resource for Qubes to function optimally.

One method of automatically shutting down a VM is to drop a .desktop file into `/home/user/.config/autostart` which execs a script that starts an app and shuts down after. There is also the `qubes-app-shutdown-idle` package that can take care of shutting down the VM (eventually).

### The solution you'd like
Add an option (preferably to GUI) that would allow for an application to be started, and the VM to shutdown after that application is done. Even better would be if this worked for a group of applications, for example, a VM that runs multiple chat programs and shuts down when all programs are finished.

### The value to a user, and who that user might be
Users who compartmentalize and have many VMs for individual tasks (which I hope is everyone) would have less cognitive load of using Qubes and experience better performance.

**Comment by deeplow** [comment]
This already happens when you open something in a disposable qube. I belive that's implemented by detecting when the program that started the qube exists.

Another way this already works as you state is with `qubes-app-shutdown-idle`. The only thing it's missing, really is not to have the [15 minutes hardcoded](https://github.com/QubesOS/qubes-app-shutdown-idle/blob/2b5e918/qubesidle/idleness_monitor.py#L26) but rather configurable as a variable. You'd want to set it to zero.

A workaround could be to have to create an autostart program that periodically checks [if it's idle ](https://github.com/QubesOS/qubes-app-shutdown-idle/blob/2b5e918fd40e7d88eb74fe4d34939c13f6274262/qubesidle/idle_watcher_window.py#L131) and if so, shuts it down.

**Comment by marmarek** [comment]
> You'd want to set it to zero (although there may be an issue where at zero seconds it hasn't even started the UI and may shut down immediately. A finite state machine to detect would fix it.)

Setting timeout to zero will have more unintended side effects. Besides shutting down before anything starts, some applications (libreoffice for example) shows a splash screen that disappears before the main window appears. If you shutdown immediately when last window is hidden, you'll shutdown at libreoffice start too. IMO there should be some non-zero timeout, even if it's small (like 1 minute).

**Comment by ben-grande** [comment]
@crat0z  Renamed the title. Please correct if wrong. The option already exists, you just want to have a shorter timeout.

**Comment by parulin** [comment]
Duplicate of https://github.com/QubesOS/qubes-issues/issues/8125? (with a related PR)

**Comment by ben-grande** [comment]
> Duplicate of [#8125](https://github.com/QubesOS/qubes-issues/issues/8125)? (with a related PR)

Yes, and I believe that issue is better worded, so closing this as duplicate.
